CIME/Tools/standard_script_setup.py

```diff
@@ -8,29 +8,35 @@
 import __main__ as main
 
 
-def check_minimum_python_version(major, minor):
+def check_minimum_python_version(major, minor, warn_only=False):
     """
     Check your python version.
 
-    >>> check_minimum_python_version(sys.version_info[0], sys.version_info[1])
+    >>> check_minimum_python_version(3, 5)
     >>>
     """
+    check = (
+        sys.version_info[0] > major or (
+            sys.version_info[0] == major and
+            sys.version_info[1] >= minor
+        )
+    )
+    if check:
+        return
     msg = (
-        "Python "
-        + str(major)
-        + ", minor version "
-        + str(minor)
-        + " is required to run CIME. You have "
-        + str(sys.version_info[0])
-        + "."
-        + str(sys.version_info[1])
+        f"Python {major}.{minor} is required to run CIME. "
```

Comment:
It's a little dangerous to use f-strings in the code that checks the python version, right? It's a lot cleaner, but old versions (3.5) don't support f-strings.

Reply:
I'd agree, probably best to stick for `format` which should cover `>=3.0`, still keeps it a little cleanish.